spot_micro_joy/scripts/spotMicroJoystickMove.py

```diff
@@ -0,0 +1,150 @@
+from geometry_msgs.msg import Twist
+from math import pi
+
+
+class SpotMicroJoystickControl():
+    BUTTON_GREEN = 0
+    BUTTON_RED = 1
+    BUTTON_YELLOW = 2
+    BUTTON_BLUE = 3
+
+    AXES_ROLL = 0
+    AXES_HEIGHT = 1
+    AXES_YAW = 2
+    AXES_PITCH = 3
+
+    WALK_AXES_FORWARD = 1
+    WALK_AXES_STRAFE = 0
+    WALK_AXES_YAW = 2
+
+    MODE_IDLE = 0
+    MODE_STAND = 1
+    MODE_ANGLE = 2
+    MODE_WALK = 3
+
+    def __init__(self):
+
+        self._angle_cmd_msg = Vector3()
+        self._angle_cmd_msg.x = 0
+        self._angle_cmd_msg.y = 0
+        self._angle_cmd_msg.z = 0
+
+        self._speed_cmd_msg = Vector3()
+        self._speed_cmd_msg.x = 0
+        self._speed_cmd_msg.y = 0
+        self._speed_cmd_msg.z = 0
+
+        self._vel_cmd_msg = Twist()
+        self._vel_cmd_msg.linear.x = 0
+        self._vel_cmd_msg.linear.y = 0
+        self._vel_cmd_msg.linear.z = 0
+        self._vel_cmd_msg.angular.x = 0
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._walk_event_cmd_msg = Bool()
+        self._walk_event_cmd_msg.data = True  # Mostly acts as an event driven action on receipt of a true message
+
+        self._stand_event_cmd_msg = Bool()
+        self._stand_event_cmd_msg.data = True
+
+        self._idle_event_cmd_msg = Bool()
+        self._idle_event_cmd_msg.data = True
+
+        rospy.loginfo("Setting Up the Spot Micro Joystick Control Node...")
+
+        # Set up and title the ros node for this code
+        rospy.init_node('spot_micro_joystick_control')
+
+        # Create publishers for commanding velocity, angle, and robot states
+        self._ros_pub_angle_cmd = rospy.Publisher('/angle_cmd', Vector3, queue_size=1)
+        self._ros_pub_vel_cmd = rospy.Publisher('/cmd_vel', Twist, queue_size=1)
+        self._ros_pub_walk_cmd = rospy.Publisher('/walk_cmd', Bool, queue_size=1)
+        self._ros_pub_stand_cmd = rospy.Publisher('/stand_cmd', Bool, queue_size=1)
+        self._ros_pub_idle_cmd = rospy.Publisher('/idle_cmd', Bool, queue_size=1)
+
+        rospy.loginfo("Joystick control node publishers corrrectly initialized")
+
+    def reset_all_motion_commands_to_zero(self):
+        '''Reset body motion cmd states to zero and publish zero value body motion commands'''
+
+        self._vel_cmd_msg.linear.x = 0
+        self._vel_cmd_msg.linear.y = 0
+        self._vel_cmd_msg.linear.z = 0
+        self._vel_cmd_msg.angular.x = 0
+        self._vel_cmd_msg.angular.y = 0
+        self._vel_cmd_msg.angular.z = 0
+
+        self._ros_pub_vel_cmd.publish(self._vel_cmd_msg)
+
+    def reset_all_angle_commands_to_zero(self):
+        '''Reset angle cmd states to zero and publish them'''
+
+        self._angle_cmd_msg.x = 0
+        self._angle_cmd_msg.y = 0
+        self._angle_cmd_msg.z = 0
+
+        self._ros_pub_angle_cmd.publish(self._angle_cmd_msg)
+
+    def on_joy(self, msg):
+        if msg.buttons[self.BUTTON_GREEN] == 1:
```

Comment:
I would recommend changing these to if, elif, elif... statements so only one can be triggered at a time. I'm not familiar with the joystick control node, but I'd want to avoid sending two commands near simultaneously if two button presses can be true at one time.

Reply:
mmmm ... i unterstand your point. my conflict is that a joystick is designed to handle multi button presses – but that makes no sense in this case. I will change it.

Reply:
Yea, ideally the control node should be able to robustly handle any control input, it's up to you if you want to change it.

Reply:
i changed it